dvc/stage/cache.py

```diff
@@ -144,9 +149,9 @@ def save(self, stage):
         COMPILED_LOCK_FILE_STAGE_SCHEMA(cache)
 
         path = self._get_cache_path(cache_key, cache_value)
-        dpath = os.path.dirname(path)
-        makedirs(dpath, exist_ok=True)
-        dump_yaml(path, cache)
+        tmp = tempfile.NamedTemporaryFile(delete=False).name
```

Comment:
These are very tiny files, so using generic /tmp and then moving to cache is fine.